src/modules/general/pwr-skill-version-info_module.tsx

```diff
@@ -0,0 +1,159 @@
+        super(props);
+        this.state = {
+            newVersionText: '',
+            anchorEl: null,
+        };
+    }
+
+    static mapStateToProps(state: ApplicationState, localProps: PwrSkillVersionInfoLocalProps): PwrSkillVersionInfoProps {
+        const initials = state.profileStore.consultant != null ? state.profileStore.consultant.initials : 'error';
+        const versions = state.skillVersionStore.currentVersions;
+        const skillId = state.skillVersionStore.serviceSkillId;
+        return {
+            initials: initials,
+            allVersions: versions,
+            serviceSkillId: skillId,
+        };
+    }
+
+    static mapDispatchToProps(dispatch: redux.Dispatch<ApplicationState>): PwrSkillVersionInfoDispatch {
+        return {
+            addNewVersion: (skillId: number, newVersion: string) => dispatch(SkillVersionActionCreator.AsyncAddSkillVersion(skillId, newVersion)),
+            loadVersions: (skillName: string) => dispatch(SkillVersionActionCreator.AsyncGetSkillVersions(skillName)),
+            deleteVersion: (skillId: number, versionToDelete: string) => dispatch(SkillVersionActionCreator.AsyncDeleteSkillVersion(skillId, versionToDelete))
+        };
+    }
+
+    componentDidMount(): void {
+        this.loadSkillData();
+    }
+
+    componentDidUpdate(prevProps: Readonly<PwrSkillVersionInfoProps & PwrSkillVersionInfoLocalProps & PwrSkillVersionInfoDispatch>,
+                       prevState: Readonly<PwrSkillVersionInfoState>, snapshot?: any): void {
+        if (this.props.skillName != prevProps.skillName || this.props.profileVersions != prevProps.profileVersions) {
+            this.loadSkillData();
+        }
+    }
+
+    private loadSkillData = () => {
+        this.props.loadVersions(this.props.skillName);
+    };
+
+
+    private handleAddNewVersion = () => {
+        if (this.props.serviceSkillId >= 0 && this.state.newVersionText) {
+            this.props.addNewVersion(this.props.serviceSkillId, this.state.newVersionText);
+            this.setState({
+                anchorEl: null,
+                newVersionText: ''
+            });
+        }
+    };
+
+    private handleNewVersionClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        this.setState({
+            anchorEl: event.currentTarget
+        });
+    };
+
+    private handleDelete = (versionName: string) => {
+        if (this.props.serviceSkillId >= 0 && versionName) {
+            this.props.deleteVersion(this.props.serviceSkillId, versionName);
+        }
+    };
+
+    private handleEnterButton = (event: KeyboardEvent) => {
+        if (event.key == 'Enter') {
+            this.setState({anchorEl: null});
+            this.handleAddNewVersion();
+        }
+    };
+
+    private mapVersions = () => {
+        let result: JSX.Element[];
+        if (this.props.handleVersionToggle && this.props.profileVersions) {
```

Comment:
this.props.handleVersionToggle ist immer definiert. this.props.profileVersions scheint auch immer truthy zu sein. Jedenfalls hab ich den else fall nicht so richtig sehen können.

Kann das also sein, dass der Überflüssig ist? Delete wirkt auch eher wie ein Admin Feature.

Wenn man das entfernen kann, dann würde ich das so machen:

```tsx
    private existingVersionChips = () => {
        return this.props.allVersions
            .sort((a, b) => a.localeCompare(b))
            .map((value, index) =>
                <Chip key={index}
                      color={this.props.profileVersions.indexOf(value) >= 0 ? 'primary' : 'default'}
                      label={value}
                      onClick={() => this.props.handleVersionToggle(value)}
                />);
    };

    private mapVersions = () => {
        return [
            ...this.existingVersionChips(),
            <Chip key={'newVersion'} icon={<AddIcon/>} label={'Neu'} color={'secondary'} onClick={this.handleNewVersionClick}/>
        ];
    };
```